lazyscribe/repository.py

```diff
@@ -372,6 +372,10 @@ def _search_artifact_versions(
                     "Please provide ``exact`` or ``asof`` as the value for ``match``"
                 ) from None
         else:
+            if match != "exact":
+                raise ValueError(
+                    "Please provide ``exact`` as the value for ``match`` as of ``version`` is an integer"
+                ) from None
```

Comment:
I'm not sure this should raise an error. If a user provides an integer version with `match="asof"`, we can provide a valid output without additional logic. I think a warning is more appropriate here

Reply:
Ok, but then correct the documentation, please.